src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -112,12 +128,56 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         )
         check_cloud_sync = asyncio.ensure_future(rclone_check_progress(job, proc))
         await proc.wait()
+        await asyncio.wait_for(check_cloud_sync, None)
+
+        if snapshot:
+            await middleware.call("zfs.snapshot.remove", snapshot)
+
         if proc.returncode != 0:
-            await asyncio.wait_for(check_cloud_sync, None)
             raise ValueError("rclone failed")
+
+        await run_script(job, cloud_sync["post_script"], "Post-script")
+
         return True
 
 
+async def run_script(job, hook, name):
+    hook = hook.strip()
+    if not hook:
+        return
+
+    if not hook.startswith("#!"):
+        hook = f"#!/bin/bash\n{hook}"
+
+    fd, name = tempfile.mkstemp()
+    os.close(fd)
```

Comment:
Why use mkstemp if you will have to close the fd and open it again with open() ?

We can use `with tempfile.NamedTemporaryFile() as f:`

Reply:
But on a second thought. Is the database the right place for this to live?

I could think of some very complicated/long hooks for this, wouldn't making it a path to a script make more sense? Thoughts?

Reply:
> We can use with tempfile.NamedTemporaryFile() as f:

When this file is open, it can't be executed (`Text file busy`). When we close it, it is automatically deleted. That's why I've decided not to use it.

Reply:
> I could think of some very complicated/long hooks for this, wouldn't making it a path to a script make more sense? Thoughts?

I think this is great when you can just type a few commands right in the box and don't think where to store scripts. For long hooks user can just type `exec /mnt/datascript/path`. We can automate this through UI (we still don't have a file browser though and I wonder why is that)

Reply:
How can UI automate this? It would have to parse the input, thats far from optimal. 

Path to script is how others UI already work, e.g. pre/post init scripts.

Reply:
UI can automate this by typing script path into box instead of user

> Path to script is how others UI already work, e.g. pre/post init scripts.

initshutdownscript also allows user to type command. And I can imagine lots of users abusing it by typing `sh -c 'cmd1; cmd2; cmd3'`. Won't it be better to have a convenient multiline input? My guess is that most hooks would be a 3-5 lines scripts.